sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregator/AbstractAggregator.java

```diff
@@ -13,24 +16,17 @@
   private volatile boolean hasRecordings = false;
 
   @Override
-  public void mergeToAndReset(Aggregator other) {
-    if (!this.getClass().isInstance(other)) {
-      return;
+  @Nullable
+  public Accumulation toAccumulationThenReset() {
+    if (!hasRecordings) {
+      return null;
     }
-    doMergeAndReset(other);
     hasRecordings = false;
+    return doToAccumulationThenReset();
   }
 
-  /**
-   * Merges the current value into the given {@code aggregator} and resets the current value in this
-   * {@code Aggregator}.
-   *
-   * <p>If this method is called, you can assume that the passed in aggregator can be cast to your
-   * self-type.
-   *
-   * @param aggregator The aggregator to merge with.
-   */
-  abstract void doMergeAndReset(Aggregator aggregator);
+  /** Implementation of the {@code toAccumulationThenReset}. */
+  abstract Accumulation doToAccumulationThenReset();
```

Comment:
urgh. this method name is getting ugly. we should think up a better name at some point.